Sources/MapboxCoreNavigation/Logger+Internal.swift

```diff
@@ -0,0 +1,50 @@
+import Foundation
+import OSLog
+@_implementationOnly import MapboxCommon_Private.MBXLog_Internal
+
+/// :nodoc:
+public struct Log {
+    public typealias Category = NavigationSdkLogCategory
+    private typealias Logger = MapboxCommon_Private.Log
+
+    public static func debug(_ message: String, category: Category) {
+        Logger.debug(forMessage: message, category: category.rawLogCategory)
+    }
+
+    public static func info(_ message: String, category: Category) {
+        Logger.info(forMessage: message, category: category.rawLogCategory)
+    }
+
+    public static func warning(_ message: String, category: Category) {
+        Logger.warning(forMessage: message, category: category.rawLogCategory)
+    }
+
+    public static func error(_ message: String, category: Category) {
+        Logger.error(forMessage: message, category: category.rawLogCategory)
+    }
+
+    public static func fault(_ message: String, category: Category) {
+        let faultLog: OSLog = .init(subsystem: "com.mapbox.navigation", category: category.rawValue)
+        os_log("%{public}@", log: faultLog, type: .fault, message)
+        Logger.error(forMessage: message, category: category.rawLogCategory)
+    }
+}
+
+/// :nodoc:
+public struct NavigationSdkLogCategory {
```

Comment:
Looks like it's worth implementing `RawRepresentable`
```suggestion
public struct NavigationSdkLogCategory: RawRepresentable {
```